controllers/configmap_controller.go

```diff
@@ -156,6 +159,10 @@ func (r *ConfigMapReconciler) Reconcile(ctx context.Context,
 // reconcileServices uses the data within the services ConfigMap to ensure WMCO-managed Windows services on
 // Windows Nodes have the expected configuration and are in the expected state
 func (r *ConfigMapReconciler) reconcileServices(ctx context.Context, windowsServices *core.ConfigMap) error {
+	if err := r.removeOutdatedServicesConfigMaps(ctx); err != nil {
```

Comment:
Should we have any checks before calling this function as this will be called every time reconcile function gets called? Anyways the upgrade doesn't happen frequently, right?

Reply:
What kind of check are you thinking?

This has to be done every time we reconcile the services ConfigMap, which is relatively very infrequently since it cannot be updated, and it is only created by WMCO on startup/upgrade.

Reply:
Note that we don't actually delete anything unless both:
- there is at least one ConfigMap in watchNamespace that has the name "windows-services-*" where * != current WMCO version
- all Windows nodes are up to date
And these two conditions are lazy evaluated in `removeOutdatedServicesConfigMaps()`